site/js/modules/webui/render.js

```diff
@@ -1,16 +1,25 @@
+import * as Autograder from '../autograder/base.js';
+
 import * as Assignment from './assignment.js';
 import * as Routing from './routing.js';
 import * as Util from './util.js';
 
-function makeCardObject(type = 'unknown', text = '', link = '#') {
+function makeCardObject(
+        type = 'unknown', text = '', link = '#',
+        minServerRole = Autograder.Users.SERVER_ROLE_UNKNOWN,
+        minCourseRole = Autograder.Users.COURSE_ROLE_UNKNOWN,
+        courseId = undefined) {
```

Comment:
Maybe we should make this a destructured object instead.

Or, do you think it is getting complex enough to warrant a class?
(Then it could have a validate that checks the roles.)